Resources/Prototypes/Entities/Objects/Misc/land_mine.yml

```diff
@@ -22,7 +22,21 @@
   - type: Sprite
     drawdepth: Items
     sprite: Objects/Misc/landmine.rsi
-    state: landmine
+    state: landmine-inactive
+    layers:
+      - state: landmine-inactive
+        map: [ "disarmed" ]
+        visible: true
+      - state: landmine
+        map: [ "armed" ]
+        visible: false
```

Comment:
Uh, sorry. I don't know how to do this. I have tried understanding from the [documentation](https://docs.spacestation14.com/en/ss14-by-example/making-a-sprite-dynamic.html) but failing. The visualizer system is confusing for me. Could you please explain?

Reply:
Here is an example from the ninja gloves:
![grafik](https://github.com/user-attachments/assets/5d107b3e-cbca-4688-91fe-15952d86f4e1)
Instead of making the first layer invisible and the second layer visible when armin, just change the state of the layer directly in the generic visualizer. That way you only need one layer instead of two.

Reply:
So I kinda understand this YAML but still struggling to make it actually change. The mine shows now with the `landmine-inactive`-sprite but arming it does not change it to `landmine`. And the already armed ones started showing with the `landmine-inactive`-sprite.

Reply:
The layer settings in the `SpriteComponent` (`- type: Sprite`) determine which sprite is initially spawned and the generic visualizer will update these layers when the Visuals are changed in C#.
For the armed ones you will have to overwrite the SpriteComponent to use the `landmine-inactive` state in the inherited prototype.

Reply:
Alright so now I have at least the starting state correct. How do I change the visuals in C#?

Reply:
Ah, I found the problem. I was using `ToggleVisuals.Layers` instead of `ToggleVisuals.Toggled`.